File: cpp/Problem283.cpp
/*
 * Problem: 283. Move Zeroes
 * URL: https://leetcode.com/problems/move-zeroes/
 * Difficulty: Easy
 * Tags: Array, Dutch Flag Algorithm
 */

#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
	void moveZeroes(vector<int>& nums) {
		for (int i = 0, j = 0, size = nums.size(); i < size; i++) {
			if (nums[i] != 0) {
				swap(nums[i], nums[j]);
				j++;
			}
		}
	}
};

File: python/Problem283.py
# Problem: 283. Move Zeroes
# URL: https://leetcode.com/problems/move-zeroes/
# Difficulty: Easy
# Tags: Array, Dutch Flag Algorithm

from typing import List

class Solution:
  def moveZeroes(self, nums: List[int]) -> None:
    j: int = 0
    for i in range(len(nums)):
      if nums[i] != 0:
        nums[i], nums[j] = nums[j], nums[i]
        j += 1